src/main/java/org/team1540/robot2019/drivecontrol/commands/UDPAutoLineup.java

```diff
@@ -0,0 +1,130 @@
+import org.team1540.robot2019.datastructures.threed.Transform3D;
+import org.team1540.robot2019.datastructures.twod.Twist2D;
+import org.team1540.robot2019.datastructures.utils.TrigUtils;
+import org.team1540.robot2019.odometry.TankDriveOdometryRunnable;
+import org.team1540.robot2019.subsystems.Drivetrain;
+import org.team1540.robot2019.utils.TankDriveTwist2DInput;
+import org.team1540.robot2019.vision.LimelightLocalization;
+import org.team1540.robot2019.wrappers.NavX;
+import org.team1540.robot2019.wrappers.TEBPlanner;
+import org.team1540.rooster.drive.pipeline.FeedForwardProcessor;
+import org.team1540.rooster.drive.pipeline.UnitScaler;
+import org.team1540.rooster.functional.Executable;
+
+public class UDPAutoLineup extends Command {
+
+    private final Drivetrain dt;
+    private final TEBPlanner planner;
+    private final LimelightLocalization limeLoc;
+    private final TankDriveOdometryRunnable driveOdometry;
+    private final Transform3D lastOdomToLimelight;
+    private final NavX navx;
+
+    Transform3D goal;
+    private Executable pipeline;
+    private TankDriveTwist2DInput twist2DInput;
+
+    public UDPAutoLineup(Drivetrain dt, TEBPlanner planner, LimelightLocalization limeLoc, TankDriveOdometryRunnable driveOdometry,
+        Transform3D lastOdomToLimelight, NavX navx) {
+        this.dt = dt;
+        this.planner = planner;
+        this.limeLoc = limeLoc;
+        this.driveOdometry = driveOdometry;
+        this.lastOdomToLimelight = lastOdomToLimelight;
+        this.navx = navx;
+        requires(dt);
+        twist2DInput = new TankDriveTwist2DInput(Tuning.drivetrainRadiusMeters);
+        pipeline = twist2DInput
+            .then(new FeedForwardProcessor(0, 0, 0))
+//        .then(new FeedForwardProcessor(0.27667, 0.054083,0.08694))
+            // .then((Processor<TankDriveData, TankDriveData>) tankDriveData -> new TankDriveData(tankDriveData.left, tankDriveData.right))
+            .then(new UnitScaler(Tuning.drivetrainTicksPerMeter, 10))
+            .then(dt.getPipelineOutput());
+    }
+
+    @Override
+    protected void initialize() {
+//    dt.reset();
+        dt.configTalonsForVelocity();
+
+        NetworkTable tebConfigTable = NetworkTableInstance.getDefault().getTable("TEBPlanner/Config");
+        tebConfigTable.getEntry("TEBReset").setBoolean(true);
+        tebConfigTable.getEntry("MaxVelX").setNumber(1.5);
+        tebConfigTable.getEntry("MaxVelXBackwards").setNumber(1.4);
+        tebConfigTable.getEntry("AccLimX").setNumber(1.0);
+        tebConfigTable.getEntry("MaxVelTheta").setNumber(5.0);
+        tebConfigTable.getEntry("AccLimTheta").setNumber(12.0);
+        if (limeLoc.attemptUpdatePose()) { // TODO: Make this distance tunable
+            computeAndUpdateGoal();
+        } else {
+            if (limeLoc.millisSinceLastAcquired() < 10000) {
+                updateGoal(lastOdomToLimelight);
+            } else {
+                //      Robot.leds.set(ColorPattern.RED);
```

Comment:
Remove these commented bits?

Reply:
I'm leaving these for now, we're going to add LEDs after the merge.

Reply:
OK